### PR TITLE
Added Data and Segment struct to User Struct

### PR DESCRIPTION
**Pull request by jamesadinmo**

### CODE REVIEW

File: src/v2_5/mod.rs [hunk 1]
```diff
@@ -30,6 +30,8 @@ mod source;
 mod supply_chain;
 mod user;
 mod video;
+mod data;
+mod segment;
```
Comment (paramako) [inline src/v2_5/mod.rs]: you need to add a public usage of your mods to use their structs outside of a crate

```
pub use self::data::*;
pub use self::segment::*;
```

File: src/v2_5/data.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+use serde_utils;
+use super::segment::Segment;
+
+// This object used by publishers to pass additional attributes about the user or content.
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct Data {
+    // The data extension object that contains community extensions.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    ext: Option<serde_utils::Ext>,
```
Comment (paramako) [inline src/v2_5/data.rs]: not an issue but it`s prettier to add the `ext` field as  last

File: src/v2_5/user.rs [hunk 3]
```diff
@@ -6,10 +6,22 @@
...
 pub struct User {
+    // A DSP's UID for this user (see user matching for format).
     #[serde(rename = "buyeruid", skip_serializing_if = "Option::is_none")]
     pub buyer_uid: Option<String>,
+    // This object used by publishers to pass additional attributes about the user or content.
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub data: Vec<Data>,
+    // The User Ext Object, which is used to indicate requests that contain certain user identifiers and are subject to GDPR regulations.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ext: Option<serde_utils::Ext>,
```
Comment (paramako) [inline src/v2_5/user.rs:22]: same here. not an issue but it`s prettier to add the `ext` field as  last

Comment (jamesadinmo) [inline src/v2_5/user.rs:22]: as specs

File: src/v2_5/data.rs (hunk 2 above)
Comment (jamesadinmo) [inline src/v2_5/data.rs]: as specs

File: src/v2_5/mod.rs (hunk 1 above)
Comment (jamesadinmo) [reply src/v2_5/mod.rs]: fixed

**Pull request merged**